### ISSUE TITLE
Questions

### ISSUE THREAD

**Issue by bpbond** [FoRTExperiment/ed4forte#4]
@kdorheim asks

```
Outstanding questions from the meeting (questions I was asked but could not answer)
•	What soil data is needed? 
•	What vegetation data do we need? 
•	What meteorological data do we need? How specific does it need to be? (I had a thought if we end up doing some sort of variability with climate and weather I am not sure if the weather of the different landscapes would matter as much)
 
Other things to think about 
•	What variables / story would be interesting and valuable for Lisa to investigate? 
•	What is the value of modeling as an educational tool? What can we / do we want to incorporate into the outreach deliverables? 
```

**Comment by kdorheim** [comment]
# ED Soil Needs 

So looking at `https://github.com/EDmodel/ED2/blob/master/ED/Template/Template/ED2IN` its looks like there quite a few soil variables in the INI file as well as a handful of soil data bases that can be set. However it is not clear to me what the strutuctre of the soil data bases would have to be. I wonder if this is something that people use pecan to format. 

## INI file variables 
There are quite a few options that can be set using the ini file 

I have the impression that this first set could be set up with two soil data bases, but I am not sure what the format of the soil data base data must be. 

* the number of layers
*  number of snow/water pounding layers 
* soil texture class 
* soil texture 
*  soil color (albedo relevant) 
* fraction of clay or sand 
* soil moisture 
* soil temperature off set 
* slope drainage 

Other soil variables that must be defined in the ED2 INI file. 
* soil moisture boundary condition at the bottom (is it bed rock, and aquifer ect.)


### TLDR 
These are the sections of the variables I pulled out of the ED2 INI file, they provide more technical despctions of the 
```
   !---------------------------------------------------------------------------------------!
   !  NZG    - Number of soil layers.  One value for all regions and polygons of interest. !
   !  NZS    - Maximum number of snow/water pounding layers.  One value for all regions    !
   !           and polygons of interest.  This is used only when snow is accumulating.     !
   !           If only liquid water is standing, the water will be always collapsed        !
   !           into a single layer.                                                        ! 
   !---------------------------------------------------------------------------------------!
```

```
  !---------------------------------------------------------------------------------------!
   ! ISOILFLG -- This controls how to initialise soil texture.  This must be a list with   !
   !             N_ED_REGION+N_POI elements.  The first N_ED_REGION elements correspond to !
   !             each gridded domain (from first to last).  Elements between N_ED_REGION+1 !
   !             and N_ED_REGION+N_POI correspond to the polygons of interest (from 1 to   !
   !             N_POI.  Options are:                                                      !
   !             1 -- Read in soil textural class from the files set in SOIL_DATABASE.     !
   !             2 -- Assign either the value set by NSLCON (see below) or define soil     !
   !                  texture from SLXSAND and SLXCLAY.                                    !
   !---------------------------------------------------------------------------------------!
```
``` 
   !---------------------------------------------------------------------------------------!
   ! NSLCON -- ED-2 Soil classes that the model will use when ISOILFLG is set to 2.        !
   !           Possible values are:                                                        !
   !---------------------------------------------------------------------------------------!
   !   1 -- sand                |   7 -- silty clay loam     |  13 -- bedrock              !
   !   2 -- loamy sand          |   8 -- clayey loam         |  14 -- silt                 !
   !   3 -- sandy loam          |   9 -- sandy clay          |  15 -- heavy clay           !
   !   4 -- silt loam           |  10 -- silty clay          |  16 -- clayey sand          !
   !   5 -- loam                |  11 -- clay                |  17 -- clayey silt          !
   !   6 -- sandy clay loam     |  12 -- peat                                              !
```
```
   !---------------------------------------------------------------------------------------!
   ! ISOILCOL -- LEAF-3 and ED-2 soil colour classes that the model will use when ISOILFLG !
   !             is set to 2.  Soil classes are from 1 to 20 (1 = lightest; 20 = darkest). !
   !             The values are the same as CLM-4.0.  The table is the albedo for visible  !
   !             and near infra-red.                                                       !
   !---------------------------------------------------------------------------------------!
   !                                                                                       !
   !       |-----------------------------------------------------------------------|       !
   !       |       |   Dry soil  |  Saturated  |       |   Dry soil  |  Saturated  |       !
   !       | Class |-------------+-------------| Class +-------------+-------------|       !
   !       |       |  VIS |  NIR |  VIS |  NIR |       |  VIS |  NIR |  VIS |  NIR |       !
   !       |-------+------+------+------+------+-------+------+------+------+------|       !
   !       |     1 | 0.36 | 0.61 | 0.25 | 0.50 |    11 | 0.24 | 0.37 | 0.13 | 0.26 |       !
   !       |     2 | 0.34 | 0.57 | 0.23 | 0.46 |    12 | 0.23 | 0.35 | 0.12 | 0.24 |       !
   !       |     3 | 0.32 | 0.53 | 0.21 | 0.42 |    13 | 0.22 | 0.33 | 0.11 | 0.22 |       !
   !       |     4 | 0.31 | 0.51 | 0.20 | 0.40 |    14 | 0.20 | 0.31 | 0.10 | 0.20 |       !
   !       |     5 | 0.30 | 0.49 | 0.19 | 0.38 |    15 | 0.18 | 0.29 | 0.09 | 0.18 |       !
   !       |     6 | 0.29 | 0.48 | 0.18 | 0.36 |    16 | 0.16 | 0.27 | 0.08 | 0.16 |       !
   !       |     7 | 0.28 | 0.45 | 0.17 | 0.34 |    17 | 0.14 | 0.25 | 0.07 | 0.14 |       !
   !       |     8 | 0.27 | 0.43 | 0.16 | 0.32 |    18 | 0.12 | 0.23 | 0.06 | 0.12 |       !
   !       |     9 | 0.26 | 0.41 | 0.15 | 0.30 |    19 | 0.10 | 0.21 | 0.05 | 0.10 |       !
   !       |    10 | 0.25 | 0.39 | 0.14 | 0.28 |    20 | 0.08 | 0.16 | 0.04 | 0.08 |       !
   !       |-----------------------------------------------------------------------|       !
   !                                                                                       !
   !   Soil type 21 is a special case in which we use the albedo method that used to be    !
   ! the default in ED-2.1.                                                                !
   !---------------------------------------------------------------------------------------!
```
```
   !---------------------------------------------------------------------------------------!
   !     These variables are used to define the soil properties when you don't want to use !
   ! the standard soil classes.                                                            !
   !                                                                                       !
   ! SLXCLAY -- Prescribed fraction of clay  [0-1]                                         !
   ! SLXSAND -- Prescribed fraction of sand  [0-1].                                        !
   !                                                                                       !
   !     They are used only when ISOILFLG is 2, both values are between 0. and 1., and     !
   ! their sum doesn't exceed 1.  In case ISOILFLG is 2 but the fractions do not meet the  !
   ! criteria, ED-2 uses NSLCON instead.                                                   !
   !---------------------------------------------------------------------------------------!
```

```
   !---------------------------------------------------------------------------------------!
   !    Soil grid and initial conditions in case no file is provided. Provide NZG values   !
   ! for the following variables (always from deepest to shallowest layer).                !
   !                                                                                       !
   ! SLZ     - depth of the bottom of each soil layer [m].  Values must be negative.       !
   ! SLMSTR  - this is the initial soil moisture, now given as the soil moisture index.    !
   !              -1 = dry air soil moisture                                               !
   !               0 = wilting point                                                       !
   !               1 = field capacity                                                      !
   !               2 = porosity (saturation)                                               !
   !           Values can be fraction, in which case they will be linearly interpolated    !
   !           between the special points (e.g. 0.5 will put soil moisture half way        !
   !           between the wilting point and field capacity).                              !
   ! STGOFF  - initial temperature offset (soil temperature = air temperature + offset)    !
   !---------------------------------------------------------------------------------------!
```
```

   !---------------------------------------------------------------------------------------!
   ! ISOILBC -- This controls the soil moisture boundary condition at the bottom.  Choose  !
   !            the option according to the site characteristics.                          !
   !            0.  Flat bedrock.  Flux from the bottom of the bottommost layer is zero.   !
   !            1.  Gravitational flow (free drainage).  The flux from the bottom of the   !
   !                bottommost layer is due to gradient of height only.                    !
   !            2.  Lateral drainage.  Similar to free drainage, but the gradient is       !
   !                reduced by the slope not being completely vertical.  The reduction is  !
   !                controlled by variable SLDRAIN.  In the future options 0, 1, and 2 may !
   !                be combined into a single option.                                      !
   !            3.  Aquifer.  Soil moisture of the ficticious layer beneath the bottom is  !
   !                always at saturation.                                                  !
```
```
   !---------------------------------------------------------------------------------------!
   ! SLDRAIN -- This is used only when ISOILBC is set to 2.  In this case SLDRAIN is the   !
   !            equivalent slope that will slow down drainage.  If this is set to zero,    !
   !            then lateral drainage reduces to flat bedrock, and if this is set to 90,   !
   !            then lateral drainage becomes free drainage.  SLDRAIN must be between 0    !
   !            and 90.                                                                    !
```

**Comment by kdorheim** [comment]
# ED Vegetation Needs
> What vegetation data do we need? 

Here it seems like there are lots options that depend on structural options, but I am pretty sure that based off Shiklomanov et al. pre print at GBC we are going to opt for the default configuration of ED and focus on the vegetation parameters. 

We will want some sort of inventory for the types of PFTs to include the in the simulations See the PFT table for our options. 
```
   !---------------------------------------------------------------------------------------!
   !      The following variables control the plant functional types (PFTs) that will be   !
   ! used in this simulation.                                                              !
   !                                                                                       !
   ! INCLUDE_THESE_PFT -- which PFTs to be considered for the simulation.                  !
   ! PASTURE_STOCK     -- which PFT should be used for pastures                            !
   !                      (used only when IANTH_DISTURB = 1 or 2)                          !
   ! AGRI_STOCK        -- which PFT should be used for agriculture                         !
   !                      (used only when IANTH_DISTURB = 1 or 2)                          !
   ! PLANTATION_STOCK  -- which PFT should be used for plantation                          !
   !                      (used only when IANTH_DISTURB = 1 or 2)                          !
   !                                                                                       !
   ! PFT table                                                                             !
   !---------------------------------------------------------------------------------------!
   ! 1 - C4 Grass                                                                          !
   ! 2 - Tropical broadleaf, early successional                                            !
   ! 3 - Tropical broadleaf, mid-successional                                              !
   ! 4 - Tropical broadleaf, late successional                                             !
   ! 5 - Temperate C3 grass                                                                !
   ! 6 - Northern North American pines                                                     !
   ! 7 - Southern North American pines                                                     !
   ! 8 - Late-successional North American conifers                                         !
   ! 9 - Temperate broadleaf, early successional                                           !
   ! 10 - Temperate broadleaf, mid-successional                                            !
   ! 11 - Temperate broadleaf, late successional                                           !
   ! 12 - (Beta) Tropical broadleaf, early successional (thick bark)                       !
   ! 13 - (Beta) Tropical broadleaf, mid-successional (thick bark)                         !
   ! 14 - (Beta) Tropical broadleaf, late successional (thick bark)                        !
   ! 15 - Araucaria                                                                        !
   ! 16 - Tropical/subtropical C3 grass                                                    !
   ! 17 - (Beta) Lianas                                                                    !
```


### Data Base 
ED reads in a vegetation data base, but it sounds like it might just be a land/water mask? 
```
   !---------------------------------------------------------------------------------------!
   !  Input databases                                                                      !
   !  VEG_DATABASE     -- vegetation database, used only to determine the land/water mask. !
   !                      Fill with the path and the prefix.                               !
```

When I plot the UMBS vegetation data base we are using right now it looks something like
I have no idea what the axis are for but does that kinda look like a map of UMBS? I'm curious. 
<img width="961" alt="Screen Shot 2020-03-19 at 3 01 09 PM" src="https://user-images.githubusercontent.com/27299759/77104619-ab96a480-69f2-11ea-9222-707be14ccc3e.png">

**Comment by kdorheim** [comment]
# ED Meteorological Data Needs 
> What meteorological data do we need? 

The meteorological driver data is read as a series of formatted hdf5 files. Looking at one of the hdf5 files it looks like to me that all of the variables are regional or for the patch that is being modeled, this data is not gridded but is a time series. 
```
   !  ED_MET_DRIVER_DB -- File containing information for meteorological driver            !
   !                      instructions (the "header" file).                                !
```

Theses files were formatted by PECAN, each individual hdf5 files contains data for a single year. 
```
# VARIABLE NAMES FOLLOW NCEP NAMING CONVENTIONS:
# nbdsf:  near IR beam downward solar radiation [W/m2]
# nddsf:  near IR diffuse downward solar radiation [W/m2]
# vbdsf:  visible beam downward solar radiation [W/m2]
# vddsf:  visible diffuse downward solar radiation [W/m2]
# prate:  precipitation rate [kg_H2O/m2/s]
# dlwrf:  downward long wave radiation [W/m2]
# pres: pressure [Pa]
# hgt: geopotential height [m]
# ugrd: zonal wind [m/s]
# vgrd: meridional wind [m/s]
# sh: specific humidity [kg_H2O/kg_air]
# tmp: temperature [K]
# co2: surface co2 concentration [ppm]
# lat: grid of latitude coordinates, if this variable is present line 3 is ignored
# lon: grid of longitude coordinates, if this variable is present line 3 is ignored
```

**Comment by bpbond** [comment]
>does that kinda look like a map of UMBS

😕 no

**Comment by kdorheim** [comment]
:( interesting

**Comment by kdorheim** [comment]
Regarding our questions about the `VEG_DATABASE` I do not think that it is relevant to the sorts of runs we are trying to do for the FoRTE experiment. 

[the following is from the ED slack community]
```

Marcos Longo  7 minutes ago
VEG_DATABASE is only useful for gridded, regional simulations, when it's used to identify and ignore polygons that are completely on water.
If you are running single-point simulations (i.e., if you set N_ED_REGION=0 in your ED2IN), then VEG_DATABASE is ignored, and the polygon is assumed to be 100% land.
```
